src/boot/efi/xbootldr.c

```diff
@@ -7,6 +7,11 @@
 #include "xbootldr.h"
 #include "util.h"
 
+union GptHeaderBuffer {
+        EFI_PARTITION_TABLE_HEADER gpt_header;
+        uint8_t space[((sizeof(EFI_PARTITION_TABLE_HEADER) + 511) / 512) * 512];
```

Comment:
one of those days we should probably move something like ALIGN_TO() into the fundamental headers and use that here

Reply:
The ALIGN_TO function is kind of magic to me. If you say it does the same I don't mind turning it into a macro and moving it into fundamental headers.

Reply:
Turning it into a constant macro should be doable with some __builtin_ magic, no?